controllers/workspace/metrics/metrics.go

```diff
@@ -21,14 +21,26 @@ const (
 	workspaceSourceLabel     = "controller.devfile.io/devworkspace-source"
 	metricSourceLabel        = "source"
 	metricsRoutingClassLabel = "routingclass"
+	metricsReasonLabel       = "reason"
+
+	reasonInfrastructureFailure = "infrastructure_failure"
+	reasonUnknown               = "unknown"
 )
 
+var infrastructureFailures = []string{
+	"CrashLoopBackOff",
+	"ImagePullBackOff",
```

Comment:
The same is here as above like typically we face this issue when CRW bits are broken and plugin registry references non-existing plugin tag. Or also user specifies non-existing image.

So, I think we should better describe what is infrastructure error (like something user can do nothing by configuring devworkspace), and what exactly we expect from each of failures.

Reply:
@l0rd WDYT? how should we handle this situation? I have raised similar concerns in https://github.com/eclipse/che/issues/20261#issuecomment-916152162

Reply:
> I have raised similar concerns in eclipse/che#20261 (comment)

yeah, sorry for losing that request and did not provide the answer from my end.

Reply:
no worries, the easiest solution would be to have a reason per failed event e.g. `reason=image_pull_back_off`, but I do not think we can afford this from the cardinality perspective

Reply:
ImagePullBackOff is a request problem. Even if that's a registry misconfiguration that's still a request problem.

Reply:
so, you mean it should be under `bad_request` reason, right? 
in this case should we put`"CrashLoopBackOff", "ImagePullBackOff"  under `bad_request`, and `CreateContainerError", "RunContainerError", "ErrImagePull" under `infrastructure_failure`?